### ISSUE TITLE
Pattern for actions that disconnect you from Cockpit

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#1818]
There's an entire class of actions that would disconnect you from Cockpit
- Stopping cockpit.service
- Changing the firewall to block the port
- Changing the IP address.

We need a decent design (maybe a dialog) for preventing or strongly warning about these actions.

**Comment by andreasn** [comment]
Will this look for the IP address you're currently connected to, if I have two or three ip addresses, due to several network cards?

**Comment by andreasn** [comment]
At the very least, we can do something like this:
![Mockup](https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/misc/disconnect-action.png)

We could also add btn-danger to the stop button on the cockpit.service page and a little info text in the ipv4/ipv6 configure dialog.

**Comment by andreasn** [comment]
![screenshot from 2015-04-15 16 02 36](https://cloud.githubusercontent.com/assets/209895/7160407/25de925c-e389-11e4-922f-ef429eec5334.png)

**Comment by stefwalter** [comment]
For netwoking, the whole approach needs a lot of discussion. Maybe including things like NetworkManager rollback if the browser disconnects right after a change?

**Comment by stefwalter** [comment]
Sorta related: networkd should be including rollback support in its API.

**Comment by Gundersanne** [comment]
So the network page has the rollback thing now.

For the firewall page this is in the works #12999.

And I'll look into doing this for disabling the cockpit socket after #1299 gets resolved. So we can do sort of the same thing there as for the services.

I'll assign this to myself

**Comment by garrett** [comment]
This is a very open ended issue with a lot of fixes that have already been made.

Are there any places that still disconnect in Cockpit without a warning? If so, please re-open with the example or — better yet — file another issue that's more specific. Thanks! :sparkles:
